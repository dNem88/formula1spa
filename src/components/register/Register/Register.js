import React, { useState, useRef, useContext, Fragment } from 'react';
import userContext from '../../../context/userContext'
import {useNavigate} from 'react-router-dom'
import styles from './register.module.css'

function Register() {
    let context = useContext(userContext);
    const navigate = useNavigate()
    const [formdata, setFormdata] = useState({
        username: '',
        email: '',
        password: '',
        confirmPassword: '',
        errorMessage: ''
    })
    const submitButton = useRef()
    function changeHandler(e) {
        setFormdata({
            ...formdata, [e.target.id]: e.target.value, errorMessage: ''
        })
    }
    async function submitHandler(e) {
        e.preventDefault();
        submitButton.current.disabled = true;
        try {
            let response = await context.register(formdata)
            console.log(response)
        if (response.error) {
            setFormdata({...formdata, errorMessage: response.error.message})
            submitButton.current.disabled = false;
        } 
        if (response.acknowledged && response.insertedId) {
            navigate('/auth/login')
        }
        } catch(e) {
            setFormdata({
                ...formdata,
                errorMessage: 'Unexpected ServerError'
            })
        }
    }
    return (
        <form onSubmit = {
            submitHandler
        } >
            <div>
                <h2>CREATE ACCOUNT</h2>
            </div>
            <label htmlFor={'username'}>Username</label>
            <input type='text' id='username' placeholder='username' value={formdata.username} onChange={changeHandler}></input>
            <label htmlFor={'email'}>Email address</label>
            <input type='text' id='email' placeholder='email' value={formdata.email} onChange={changeHandler}></input>
            <label htmlFor={'password'}>Password</label>
            <input type='password' id='password' placeholder='password' value={formdata.password} onChange={changeHandler}></input>
            <label htmlFor={'confirmPassword'}>Confirm password</label>
            <input type='password' id='confirmPassword' placeholder='confirm password' value={formdata.confirmPassword} onChange={changeHandler}></input>
            <section>
                <p className={styles.error}>{formdata.errorMessage}</p>
                <p >Password must contain</p>
                <ul>
                    <li>Uppercase</li>
                    <li>Lowercase (at least 2)</li>
                    <li>8 or more characters</li>
                    <li>Number</li>
                </ul>
            </section>
            <button className={styles.button} type='submit' ref={submitButton}>REGISTER</button>
        </form>
    )
}

export default Register