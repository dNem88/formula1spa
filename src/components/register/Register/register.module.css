form{
    width: clamp(22rem, 40vw, 40rem);
    padding: 2rem;
    margin-top: 3rem;

    display: flex;
    flex-direction: column;
    background-color: #e4e1dd;
    border: .7px solid #333;
    border-radius: .8rem;
}
h2{
    color: #333;
    padding-top: 0;
    margin-top: 0;
    font-style: bold;
    margin-bottom: 0;
}
input{
    height: 7vh;
    border-radius: .8rem;
    border-color: #333;
    border-width: .7px;
    padding-left: 1rem;
    font-size: clamp(1rem, .8vw, 2rem);
}
label{
    margin-top: 3vh;
    padding: .85vh;
    color: #333;
    font-size: clamp(1rem, .8vw, 2rem);
}
section{
    color: #333;
    font-size: clamp(1rem, .8vw, 2rem);
}
.button{
    font-size: clamp(1rem, .8vw, 1.3rem);
    padding: .5rem;
    color: white;
    background-color: var(--color-red);
    width: clamp(7.5rem, 11vw, 15rem);
    border-radius: .8rem;
    border-color: transparent;
    cursor: pointer;
    align-self: flex-end;
}
.button:hover{
    border-color: var(--color-red);
    color: var(--color-red);
    background-color: white;
}
.error{
    color: var(--color-red);
    padding: 0;
}