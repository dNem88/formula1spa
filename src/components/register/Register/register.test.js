import renderer from 'react-test-renderer';
import Register from './Register'
import React from 'react'
import userContext from '../../../context/userContext'
import {
    MemoryRouter as Router
} from 'react-router-dom'
describe('test  Register component', () => {
  it('Should render Register form', () => {
    let registerForm = renderer.create(
        <Router>
            <userContext.Provider value={{isLogged: false}}>
                <Register />
            </userContext.Provider>
        </Router>
    ).toJSON()
    expect(registerForm).toMatchSnapshot()
    expect(registerForm.type).toBe('form')
    expect(registerForm.children[0].children[0].type).toBe('h2')
    expect(registerForm.children[0].children[0].children[0]).toEqual('CREATE ACCOUNT')
    let usernameLabel1 = registerForm.children[1]
    let usernameInput1 = registerForm.children[2]
    expect(usernameLabel1.props.htmlFor).toEqual(usernameInput1.props.id)
    expect(usernameInput1.type).toBe('input')
    expect(usernameLabel1.type).toBe('label')
    let emailLabel2 = registerForm.children[3]
    let emailInput2 = registerForm.children[4]
    expect(emailLabel2.props.htmlFor).toEqual(emailInput2.props.id)
    expect(emailInput2.type).toBe('input')
    expect(emailLabel2.type).toBe('label')
    let passwordLabel3 = registerForm.children[5]
    let passwordInput3 = registerForm.children[6]
    expect(passwordLabel3.props.htmlFor).toEqual(passwordInput3.props.id)
    expect(passwordInput3.type).toBe('input')
    expect(passwordLabel3.type).toBe('label')
    let confirmPasswordLabel4 = registerForm.children[7]
    let confirmPasswordInput4 = registerForm.children[8]
    expect(confirmPasswordLabel4.props.htmlFor).toEqual(confirmPasswordInput4.props.id)
    expect(confirmPasswordInput4.type).toBe('input')
    expect(confirmPasswordLabel4.type).toBe('label')
  })
  
})