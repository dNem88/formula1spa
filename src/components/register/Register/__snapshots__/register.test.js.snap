// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test  Register component Should render Register form 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <h2>
      CREATE ACCOUNT
    </h2>
  </div>
  <label
    htmlFor="username"
  >
    Username
  </label>
  <input
    id="username"
    onChange={[Function]}
    placeholder="username"
    type="text"
    value=""
  />
  <label
    htmlFor="email"
  >
    Email address
  </label>
  <input
    id="email"
    onChange={[Function]}
    placeholder="email"
    type="text"
    value=""
  />
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    id="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <label
    htmlFor="confirmPassword"
  >
    Confirm password
  </label>
  <input
    id="confirmPassword"
    onChange={[Function]}
    placeholder="confirm password"
    type="password"
    value=""
  />
  <section>
    <p
      className="error"
    >
      
    </p>
    <p>
      Password must contain
    </p>
    <ul>
      <li>
        Uppercase
      </li>
      <li>
        Lowercase (at least 2)
      </li>
      <li>
        8 or more characters
      </li>
      <li>
        Number
      </li>
    </ul>
  </section>
  <button
    className="button"
    type="submit"
  >
    REGISTER
  </button>
</form>
`;
